### PR TITLE
Update fine tuning tutorial

### PR DESCRIPTION
**Pull request by albertov19**
### Description

Massive overhaul to the fine-tuning notebook, not using the helper functions, and doing a deeper analysis on things.

### Checklist

- [x] **Required** - I have added a label to this PR 🏷️
- [x] **Required** - I have run my changes through Grammarly
- [ ] If pages have been moved, I have created redirects in the `kluster-mkdocs` repo

### CODE REVIEW

File: tutorials/klusterai-api/finetuning-sent-analysis.ipynb [hunk 1]
```diff
@@ -308,36 +510,43 @@
...
     "\n",
-    "If you specify a different base model, the fine-tuning job will fail. You can also tweak the hyperparameters (such as number of epochs, batch size, and learning rate) to adjust training time and potential performance gains. Remember that increasing the number of epochs will lead to longer training time but may result in higher performance. If you're unsure which hyperparameters to set, you can also comment them out to accept the default values."
+    "Optionally, you can also provide an object containing the <a href=\"/api-reference/reference/#create-a-fine-tuning-job\" target=\"_blank\">hyperparameters for fine-tuning</a>, such as number of epochs, batch size, and learning rate, to adjust training time and potential performance gains. Remember that increasing the number of epochs will lead to longer training time but may result in higher performance. If you're unsure which hyperparameters to set, you can also comment them to accept the default values."
```
Comment (themacexpert) [inline tutorials/klusterai-api/finetuning-sent-analysis.ipynb:544]: We should have this as a full link https://docs.kluster.ai/api-reference/reference/ for example, because it will not work from the Jupyter Notebook - gets 404.

File: tutorials/klusterai-api/finetuning-sent-analysis.ipynb [hunk 2]
```diff
@@ -308,36 +510,43 @@
...
-   },
+   "id": "9716bdd7-ba56-47ab-81e9-f12f2c29dbb8",
+   "metadata": {},
    "source": [
-    "Next, we'll submit the job to the kluster.ai fine-tuning API. Currently, two base models are supported for fine-tuning:\n",
-    "- klusterai/Meta-Llama-3.1-8B-Instruct-Turbo\n",
-    "- klusterai/Meta-Llama-3.3-70B-Instruct-Turbo\n",
+    "### Start the fine-tuning \n",
+    "\n",
+    "Once the fine-tuning file has been successfully uploaded, we're ready to start the fine-tuning job by providing the file ID we got in the previous step. To do so, we use the `fine_tuning.jobs.create` method, for which we also need to provide the model to fine-tune (check the <a href=\"/get-started/models/\" target=\"_blank\">Supported models</a> section). \n",
```
Comment (themacexpert) [inline tutorials/klusterai-api/finetuning-sent-analysis.ipynb:542]: We should have this as a full link, e.g., https://docs.kluster.ai/get-started/models/ otherwise gets 404 from Jupyter notebook.

File: tutorials/klusterai-api/finetuning-sent-analysis.ipynb [hunk 3]
```diff
@@ -517,70 +742,117 @@
...
+    "Now you can submit <a href=\"/get-started/start-building/real-time/\" target=\"_blank\">real-time</a>\n",
+    " or <a href=\"/get-started/start-building/batch/\" target=\"_blank\">batch</a> inference by providing your fine-tuned model's name as the `model` for the inference call.\n",
+    "\n",
+    "In the next section, we will perform batch inference against the testing data points and compare the accuracy against a larger model.\n",
+    "\n",
+    "## Test the fine-tuned model with batch inference\n",
+    "\n",
+    "In this section, we’ll run batch requests to the fine-tuned and baseline models, using the data points left over for testing.\n",
+    "\n",
+    "For the batch inference, we will follow a similar pipeline to that of the <a href=\"/tutorials/klusterai-api/model-comparison/#perform-batch-inference\" target=\"_blank\">model comparison notebook</a>.\n",
```
Comment (themacexpert) [inline tutorials/klusterai-api/finetuning-sent-analysis.ipynb:754]: use full link - 404

File: tutorials/klusterai-api/model-comparison.ipynb [hunk 4]
```diff
@@ -1,909 +1,909 @@
...
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "937ef87f-cd16-44f1-b76a-97af7c999e0b",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{\"custom_id\": \"DeepSeek-V3-0324-0-analysis\", \"method\": \"POST\", \"url\": \"/v1/chat/completions\", \"body\": {\"model\": \"deepseek-ai/DeepSeek-V3-0324\", \"temperature\": 0.5, \"messages\": [{\"role\": \"system\", \"content\": \"\\n    You are a helpful assitant that classifies movie genres based on the movie description. Your response must be one of these exact words and nothing else: \\n    Action, Adventure, Animation, Biography, Comedy, Crime, Drama, Family, Fantasy, Film-Noir, History, Horror, Music, Musical, Mystery, Romance, Sci-Fi, Sport, Thriller, War, Western.\\n    Respond with ONE WORD ONLY. No punctuation, no explanation, no formatting. Just the genre.\\n    \"}, {\"role\": \"user\", \"content\": \"Two imprisoned men bond over a number of years, finding solace and eventual redemption through acts of common decency.\"}]}}\n"
```
Comment (themacexpert) [inline tutorials/klusterai-api/model-comparison.ipynb:395]:
```suggestion
      "{\"custom_id\": \"DeepSeek-V3-0324-0-analysis\", \"method\": \"POST\", \"url\": \"/v1/chat/completions\", \"body\": {\"model\": \"deepseek-ai/DeepSeek-V3-0324\", \"temperature\": 0.5, \"messages\": [{\"role\": \"system\", \"content\": \"\\n    You are a helpful assistant that classifies movie genres based on the movie description. Your response must be one of these exact words and nothing else: \\n    Action, Adventure, Animation, Biography, Comedy, Crime, Drama, Family, Fantasy, Film-Noir, History, Horror, Music, Musical, Mystery, Romance, Sci-Fi, Sport, Thriller, War, Western.\\n    Respond with ONE WORD ONLY. No punctuation, no explanation, no formatting. Just the genre.\\n    \"}, {\"role\": \"user\", \"content\": \"Two imprisoned men bond over a number of years, finding solace and eventual redemption through acts of common decency.\"}]}}\n"
```

File: tutorials/klusterai-api/sentiment-analysis-api.ipynb
Comment (themacexpert) [inline tutorials/klusterai-api/sentiment-analysis-api.ipynb:1]: First sentence in the notebook should say connotation rather than notation.

**Review by themacexpert — CHANGES_REQUESTED** [review]

**Review by themacexpert — DISMISSED** [review]

File: tutorials/klusterai-api/model-comparison.ipynb [hunk 5]
```diff
@@ -1,909 +1,909 @@
...
+    "\n",
+    "This notebook is prepared for you to follow along. Run the cells below to watch it all come together.\n",
+    "\n",
+    "### Create the batch job file\n",
+    "\n",
+    "This example compares the DeepSeek V3 0324, Llama 4 Maverick 17B 128E, Llama 4 Scout 17B 16E, and Gemma 3 27B models. If you'd like to test different models, feel free to modify the scripts accordingly. DeepSeek R1 was excluded from the comparison, as it consistently returned verbose responses despite multiple attempts to adjust the system prompt.\n",
+    "\n",
+    "Please refer to the <a href=\"/get-started/models/#model-comparison-table\" target=\"_blank\">Supported models</a> section for a list of the models we support.\n",
+    "\n",
+    "The following snippets prepare the JSONL file, where each line represents a different request. Note that each separate batch request can have its own model. Also, we are using a temperature of `0.5` but feel free to change it and play around with the different outcomes (but we are only asking to respond with a single word, the genre)."
```
Comment (0xlukem) [inline tutorials/klusterai-api/model-comparison.ipynb:278]:
```suggestion
    "The following snippets prepare the JSONL file, where each line represents a different request. Note that each separate batch request can have its own model. Also, we are using a temperature of `0.5`, but feel free to change it and play around with the different outcomes (but we are only asking to respond with a single word, the genre)."
```

File: tutorials/klusterai-api/model-comparison.ipynb [hunk 6]
```diff
@@ -1,909 +1,909 @@
...
+    "    while not all_completed:\n",
+    "        all_completed = True\n",
+    "        output_lines = []\n",
+    "\n",
+    "        # Loop through all batch jobs\n",
+    "        for job in batch_jobs:\n",
+    "            updated_job = client.batches.retrieve(job.id)\n",
+    "            status = updated_job.status\n",
+    "\n",
+    "            # If job is completed\n",
```
Comment (0xlukem) [inline tutorials/klusterai-api/model-comparison.ipynb:581]:
```suggestion
    "            # If the job is completed\n",
```

File: tutorials/klusterai-api/model-comparison.ipynb [hunk 7]
```diff
@@ -1,909 +1,909 @@
...
+    "            if status == \"completed\":\n",
+    "                output_lines.append(\"Job completed!\")\n",
+    "            # If job failed, cancelled or expired\n",
```
Comment (0xlukem) [inline tutorials/klusterai-api/model-comparison.ipynb:584]:
```suggestion
    "            # If the job failed, cancelled, or expired\n",
```

File: tutorials/klusterai-api/model-comparison.ipynb [hunk 8]
```diff
@@ -1,909 +1,909 @@
...
+    "            elif status in [\"failed\", \"cancelled\", \"expired\"]:\n",
+    "                output_lines.append(f\"Job ended with status: {status}\")\n",
+    "                break\n",
+    "            # If job is ongoing\n",
```
Comment (0xlukem) [inline tutorials/klusterai-api/model-comparison.ipynb:588]:
```suggestion
    "            # If the job is ongoing\n",
```

File: tutorials/klusterai-api/model-comparison.ipynb [hunk 9]
```diff
@@ -1,909 +1,909 @@
...
+    "  - Best overall ratio between cost-efficiency and accuracy\n",
+    "\n",
+    "\n",
+    "Each model was evaluated using the same set of 1000 movie overviews, with genre predictions compared against annotated ground-truth genres. Total input and output tokens were recorded and converted into USD using published prices per 1M tokens.\n",
+    "\n",
+    "To submit and evaluate the batch jobs, we:\n",
+    "\n",
+    "1. Created the JSONL file, where each file line represented a separate request\n",
+    "2. Submitted the file to the platform\n",
+    "3. Started the batch job, and monitored its progress\n",
```
Comment (0xlukem) [inline tutorials/klusterai-api/model-comparison.ipynb:877]:
```suggestion
    "3. Started the batch job and monitored its progress\n",
```

**Review by 0xlukem — COMMENTED** [review]
Great work 👍 I just coments about formatting stuff

File: tutorials/klusterai-api/finetuning-sent-analysis.ipynb [hunk 10]
```diff
@@ -181,16 +216,73 @@
...
+       "      <th></th>\n",
+       "      <th>sentiment</th>\n",
+       "      <th>text</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>neutral</td>\n",
+       "      <td>According to Gran , the company has no plans to move all production to Russia , although that is where the company is growing .</td>\n",
```
[the remark this answers is not in the document]
Comment (albertov19) [reply tutorials/klusterai-api/finetuning-sent-analysis.ipynb:249]: This is part of the original dataset, we can't modify this

File: tutorials/klusterai-api/finetuning-sent-analysis.ipynb [hunk 11]
```diff
@@ -181,16 +216,73 @@
...
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>neutral</td>\n",
+       "      <td>Technopolis plans to develop in stages an area of no less than 100,000 square meters in order to host companies working in computer technologies and telecommunications , the statement said .</td>\n",
```
[the remark this answers is not in the document]
Comment (albertov19) [reply tutorials/klusterai-api/finetuning-sent-analysis.ipynb:254]: Same comment as above

File: tutorials/klusterai-api/finetuning-sent-analysis.ipynb [hunk 12]
```diff
@@ -181,16 +216,73 @@
...
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>negative</td>\n",
+       "      <td>The international electronic industry company Elcoteq has laid off tens of employees from its Tallinn facility ; contrary to earlier layoffs the company contracted the ranks of its office workers , the daily Postimees reported .</td>\n",
```
[the remark this answers is not in the document]
Comment (albertov19) [reply tutorials/klusterai-api/finetuning-sent-analysis.ipynb:259]: Same

**Review by DrZuzzjen — APPROVED** [review]
